python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -278,54 +278,84 @@ def collect_agg(self, *, depth: int) -> AggInfo:
         )
 
 
-class NamedExpr(Expr):
-    __slots__ = ("name", "children")
-    _non_child = ("dtype", "name")
+class NamedExpr:
```

Comment:
Decided to deliberately not make this one an `Expr` (it should not appear when evaluating expressions themselves, only when constructing return values in dataframe (`IR`) nodes)

Reply:
This could be helpful to leave as a comment.
```suggestion
class NamedExpr:
    # NamedExpr does not inherit from Expr because it should not appear when
    # evaluating expressions themselves, only when constructing return values
    # in dataframe (IR) nodes).
```